### PR TITLE
Kriging enhancements

### PR DESCRIPTION
**Pull request by jschueller**
- Add a flag to optimize covariance model parameters (adutfoy/control_glm_optim)
- Homogenize covariance model, spectral model parameters names and order
- Kriging observation noise
- Add Kriging samples accessors

### CODE REVIEW

**Comment by regislebrun** [comment]
+1

File: lib/src/Uncertainty/Algorithm/MetaModel/Kriging/GeneralizedLinearModelAlgorithm.cxx [hunk 1]
```diff
@@ -592,9 +592,11 @@ NumericalScalar GeneralizedLinearModelAlgorithm::computeLogLikelihood(const Nume
   // The next step is to compute the norm of this last one and added
   // rho is the residual choleskyFactor * (Y-F*beta)
   const NumericalScalar epsilon = rho_.normSquare();
+  LOGINFO(OSS(false) << "epsilon=" << epsilon);
```
Comment (sofianehaddad) [inline lib/src/Uncertainty/Algorithm/MetaModel/Kriging/GeneralizedLinearModelAlgorithm.cxx]: Maybe a debug level?

File: lib/src/Uncertainty/Algorithm/MetaModel/Kriging/GeneralizedLinearModelAlgorithm.cxx [hunk 2]
```diff
@@ -592,9 +592,11 @@ NumericalScalar GeneralizedLinearModelAlgorithm::computeLogLikelihood(const Nume
...
   if (epsilon <= 0) return SpecFunc::MaxNumericalScalar;
   // For the general multidimensional case, we have to compute the general log-likelihood (ie including marginal variances)
   logLikelihood -= 0.5 * epsilon;
+  LOGINFO(OSS(false) << "Log-likelihood=" << logLikelihood);
```
Comment (sofianehaddad) [inline lib/src/Uncertainty/Algorithm/MetaModel/Kriging/GeneralizedLinearModelAlgorithm.cxx]: useless as it is printed just after

File: lib/src/Uncertainty/Algorithm/MetaModel/Kriging/GeneralizedLinearModelAlgorithm.cxx [hunk 3]
```diff
@@ -610,6 +612,7 @@ NumericalScalar GeneralizedLinearModelAlgorithm::computeLapackLogDeterminantChol
 
   LOGINFO("Discretize the covariance model...");
   CovarianceMatrix C(model.discretize(normalizedInputSample_));
+  LOGINFO(OSS(false) << "C=\n" << C);
```
Comment (sofianehaddad) [inline lib/src/Uncertainty/Algorithm/MetaModel/Kriging/GeneralizedLinearModelAlgorithm.cxx]: I dont think that it is a good idea to show a "huge" matrix

File: lib/src/Uncertainty/Algorithm/MetaModel/Kriging/GeneralizedLinearModelAlgorithm.cxx [hunk 4]
```diff
@@ -637,21 +640,28 @@ NumericalScalar GeneralizedLinearModelAlgorithm::computeLapackLogDeterminantChol
                                          << " Scaling up to "  << cumulatedScaling << " was not enough";
   if (cumulatedScaling > 0.0)
     LOGWARN(OSS() <<  "Warning! Scaling up to "  << cumulatedScaling << " was needed in order to get an admissible covariance. ");
+  LOGINFO(OSS(false) << "covarianceCholeskyFactor_=\n" << covarianceCholeskyFactor_);
```
Comment (sofianehaddad) [inline lib/src/Uncertainty/Algorithm/MetaModel/Kriging/GeneralizedLinearModelAlgorithm.cxx]: same remark as for C

**Comment by regislebrun** [comment]
@jschueller I agree with Sofiane: we should show only information messages or small objects (NP, scalar, bool) at INFO level and keep the potentially huge objects to the debug level, else it is unreadable.

**Comment by sofianehaddad** [comment]
@jschueller @regislebrun what about kriging in dimension 1? I mean is expected to perform kriging by handling analytically sigma in dimension 1 ? (Here or other PR?)

**Comment by jschueller** [comment]
it is done in https://github.com/adutfoy/openturns/commit/51b3f1f240850be8c4eba56ba5cbcf320c9e131a, I did not resubmit it here as that part of the code did not work as is.

**Comment by sofianehaddad** [comment]
ok maybe we should perform this last one. We may merge this for the moment, +1

**Pull request merged**